### PR TITLE
Add snapshot building to MTT helloworld

### PR DESCRIPTION
**Pull request by BaptisteGerondeau**
This patch adds the building of a snapshot of OpenMPI to be built and used to perform the hello world test, in addition to OpenHPC's OpenMPI stack.

### CODE REVIEW

File: files/ansible/deploy_mtt.yml [hunk 1]
```diff
@@ -56,16 +63,23 @@
             - "{{ test_plans }}"
     when:
       - inventory_hostname in groups['sms']
-        
-  - name: Create directory because MTT
+
+  - name: Create Directory for Snapshot module template
     file:
-            path: "/opt/mtt/samples/python/"
+            path: "/opt/ohpc/pub/moduledeps/gnu8/openmpi4"
```
Comment (rengolin) [inline files/ansible/deploy_mtt.yml]: hum, this is dependent on default compiler and mpi stack, can we pick that from somewhere else?

File: files/ansible/deploy_mtt.yml [hunk 2]
```diff
@@ -56,16 +63,23 @@
...
             state: directory
     when:
       - inventory_hostname in groups['sms']
+        
+  - name: Put in the Lmod OpenMPI Snapshot module template
+    template:
+            src: "templates/ompi_snapshot_lmod.j2"
+            dest: "/opt/ohpc/pub/moduledeps/gnu8/openmpi4/4.1.0"
```
Comment (rengolin) [inline files/ansible/deploy_mtt.yml]: perhaps these names (gnu8, openmpi4, 4.1.0) should be variables in a config file?

File: files/ansible/templates/ompi_hello_world.ini.j2 [hunk 3]
```diff
@@ -28,6 +42,19 @@ plugin = Shell
 
 # Use our built ompi install
 command = mpicc /home/test/mtt/samples/python/mpi_hello_world.c -o mpi_hello_world 
+modules = openmpi3
+modules_unload = openmpi4/4.1.0
```
Comment (rengolin) [inline files/ansible/templates/ompi_hello_world.ini.j2]: Isn't this going to conflict with the module template above?

File: files/ansible/templates/ompi_snapshot_lmod.j2 [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#%Module1.0#####################################################################
+
+proc ModulesHelp { } {
+
+puts stderr " "
+puts stderr "This module loads the openmpi4 library built with the gnu8 toolchain."
+puts stderr "\nVersion 4.1.0\n"
+
+}
+module-whatis "Name: openmpi4 built with gnu8 toolchain"
```
Comment (rengolin) [inline files/ansible/templates/ompi_snapshot_lmod.j2]: these names should also come from template parameters

File: files/ansible/templates/ompi_snapshot_lmod.j2 [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+module-whatis "Description: A powerful implementation of MPI"
+module-whatis "URL: http://www.open-mpi.org"
+
+set     version			    4.1.0
+
+setenv          MPI_DIR             /home/test/mtt/mttscratch/MiddlewareBuild_OMPIMaster/
+prepend-path    PATH                /home/test/mtt/mttscratch/MiddlewareBuild_OMPIMaster/bin/
+prepend-path    MANPATH             /home/test/mtt/mttscratch/MiddlewareBuild_OMPIMaster/share/man
+prepend-path	LD_LIBRARY_PATH	    /home/test/mtt/mttscratch/MiddlewareBuild_OMPIMaster/lib
+prepend-path    MODULEPATH          /opt/ohpc/pub/moduledeps/gnu8-openmpi3
```
Comment (rengolin) [inline files/ansible/templates/ompi_snapshot_lmod.j2]: openmpi3 or 4?

File: files/ansible/deploy_mtt.yml (hunk 1 above)
Comment (BaptisteGerondeau) [reply files/ansible/deploy_mtt.yml]: AFAIK no we can't, Lmod is very picky it seems about where to look for its module files.

File: files/ansible/templates/ompi_hello_world.ini.j2 (hunk 3 above)
Comment (BaptisteGerondeau) [reply files/ansible/templates/ompi_hello_world.ini.j2]: No, here it unloads our module (so our snapshot MPI stack) and loads the OHPC one.

File: files/ansible/templates/ompi_snapshot_lmod.j2 (hunk 5 above)
Comment (BaptisteGerondeau) [reply files/ansible/templates/ompi_snapshot_lmod.j2]: openmpi3, we are reusing the exisiting ecosystem of module files.

**Comment by rengolin** [comment]
Right, as an experiment, I don't think we should be chasing super high level automation, as we have no idea what will remain and what will change. I'll merge for now and as it evolves, we can try to refactor later.

**Pull request merged**